Variable Schema Docs (#292)
DevCycleHQ/devcycle-docs: docs/home/feature-management/features-and-variables/variables-and-variations.md

A Variation may be deleted at any time by clicking the edit icon on the Variation column of the Remote Variables page. Variations that are currently being used in any **Enabled** environment cannot be deleted. First remove any audience being targeted by this Variation prior to deletion.

## Variable Schemas

> Variable Schemas is a new DevCycle feature that lets you define allowable values for your variables when creating feature variations. Variable Schemas help developers ensure that team members can easily select allowable inputs for variable values without being nervous that a certain value might cause an application crash or negatively impact end user experience. 

Reviewer: ```suggestion
Variable Schemas is a new DevCycle feature that lets you define allowable values for your feature's variations. Variable Schemas help developers ensure that team members can easily set valid inputs for variation values without the worry that a certain value might cause an application crash or negatively impact end user experience. 
```

### Defining a Variable Schema

To define a Variable Schema for a variable, navigate to the Schema section of the Variable Details page to select and define a schema for that particular variable. The variable must be unassociated in order to select and define a schema. 

There are three different types of schemas that you can define for your variable: 

- Enum (*String or Number Variable)*
- Regex (*String Variable*)
- JSON (*JSON Variable*)

> For each type of schema, you must write a **Brief Description** for the variable schema you are about to define. For example, if I was to create define an Enum schema for a String variable that controlled a user greeting message on a sign up page, a sample description could be “Greeting options for sign-up page.” 

Reviewer: ```suggestion
Each schema type requires  a **Brief Description**. For example, when defining an Enum schema for a String variable that's meant for a greeting message on a sign up page, a sample description could be “Greeting options for sign-up page.” 
```

You also must set an **Example Value**. The example value will be used as the initial value for each variation when adding this variable to a feature.

> An **Enum** schema will allow you to to input allowable values for a string or number variable. 

Reviewer: ```suggestion
An **Enum** schema will allow you to input allowable values for a string or number variable. 
```

> If “enum” type is selected, users define the allowable variable values in the **Enum Values** input field below the type dropdown.

Reviewer: ```suggestion
If “Enum” type is selected, users must define the allowable values in the **Enum Values** input field.
```

*Enum Example:*
![Enum Schema Example](/june-2023-enum-schema-example.png)

> A **Regex** schema allows you to define a regex pattern that that must be followed for the string variable values. 

Reviewer: ```suggestion
A **Regex** schema allows you to define a regex pattern. Variation values for a variable with a Regex schema type must match the pattern defined in the schema.
```

*Regex Example:* 
![Regex Schema Example](/june-2023-regex-schema-example.png)

> A **JSON** schema will allow DVC users to define a JSON schema that must be followed for the JSON variable values.  If “JSON schema” type is selected, users can define the schema below in a JSON editor. 

Reviewer: ```suggestion
A **JSON** schema will allow DVC users to define a JSON schema that variation values must follow.  If “JSON schema” type is selected, users can define the schema below in a JSON editor. Example JSON Schemas and more information about them can be found [here](https://json-schema.org/learn/miscellaneous-examples.html#basic).
```
